test/studies/shootout/k-nucleotide/benmcd/knucleotide-user-hash.chpl

```diff
@@ -0,0 +1,149 @@
+use IO, Map, Sort;
+
+config param tableSize = 2**16,
+             columns = 61;
+
+record intWrapper {
+  var val: int;
+  proc init() {
+    val = 0;
+  }
+  proc init(i: int) {
+    val = i;
+  }
+  proc hash() {
+    return val;
+  }
+}
+
+proc main(args: [] string) {
+  // Open stdin and a binary reader channel
+  const consoleIn = openfd(0),
+        fileLen = consoleIn.size,
+        stdinNoLock = consoleIn.reader(kind=ionative, locking=false);
+
+  // Read line-by-line until we see a line beginning with '>TH'
+  var buff: [1..columns] uint(8),
+      lineSize = 0,
+      numRead = 0;
+
+  while stdinNoLock.readline(buff, lineSize) && !startsWithThree(buff) do
+    numRead += lineSize;
+
+  // Read in the rest of the file
+  var dataDom = {1..fileLen-numRead},
+      data: [dataDom] uint(8),
+      idx = 1;
+
+  while stdinNoLock.readline(data, lineSize, idx) do
+    idx += lineSize - 1;
+  
+  // Resize our array to the amount actually read
+  dataDom = {1..idx};
+
+  // Make everything uppercase
+  forall d in data do
+    d -= ("a".toByte() - "A".toByte());
+
+  writeFreqs(data, 1);
+  writeFreqs(data, 2);
+  writeCount(data, "GGT");
+  writeCount(data, "GGTA");
+  writeCount(data, "GGTATT");
+  writeCount(data, "GGTATTTTAATT");
+  writeCount(data, "GGTATTTTAATTTATAGT");
+}
+
+
+proc writeFreqs(data, param nclSize) {
+  const freqs = calculate(data, nclSize);
+
+  var arr = for (s,f) in freqs.items() do (f,s.val);
```

Comment:
It seems like the comments in knucleotide3.chpl weren't preserved here (?).  I'd keep the diff as small as possible so that someone wanting to understand the differences between the two versions can focus on what's important and not what's not.